### PR TITLE
Adopt Update dialogs to the new modal widget experience

### PR DESCRIPTION
**Pull request by erikjv**
Fixes: #11593

### CODE REVIEW

**Comment by erikjv** [comment]
New message box:

Screenshot 2024-05-03 at 14 57 17

**Comment by erikjv** [comment]
Update downloaded dialog:

Screenshot 2024-05-03 at 15 06 54

**Comment by TheOneRing** [comment]
Hmm could we use the same icon, the app icon, for both?
Also I guess both should be centered and not aligned to top?

File: src/gui/updater/ocupdater.cpp [hunk 1]
```diff
@@ -376,50 +377,17 @@ void WindowsUpdater::versionInfoArrived(const UpdateInfo &info)
...
-    QLabel *ico = new QLabel;
-    ico->setFixedSize(iconSize, iconSize);
-    ico->setPixmap(infoIcon.pixmap(iconSize));
-    QLabel *lbl = new QLabel;
     QString txt = tr("<p>A new version of the %1 Client is available.</p>"
                      "<p><b>%2</b> is available for download. The installed version is %3.</p>")
                       .arg(Utility::escape(Theme::instance()->appNameGUI()),
                           Utility::escape(info.versionString()), Utility::escape(Version::versionWithBuildNumber().toString()));
+    auto *widget = new NoUrlWidget(ocApp()->gui()->settingsDialog(), txt);
+    widget->setAttribute(Qt::WA_DeleteOnClose);
```
Comment (TheOneRing) [inline src/gui/updater/ocupdater.cpp:385]: Its not a dialog anymore, we should handle its deletion.

File: src/gui/updater/nourlwidget.h [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+}
+
+namespace OCC {
+
+class NoUrlWidget : public QWidget
+{
+    Q_OBJECT
+
+public:
+    explicit NoUrlWidget(QWidget *parent, const QString &statusMessage);
```
Comment (TheOneRing) [inline src/gui/updater/nourlwidget.h:31]: What is the meaning of the class name?

File: src/gui/updater/updatedownloadeddialog.cpp [hunk 3]
```diff
@@ -44,9 +44,14 @@ UpdateDownloadedDialog::UpdateDownloadedDialog(QWidget *parent, const QString &s
     yesButton->setDefault(true);
 }
 
-UpdateDownloadedDialog::~UpdateDownloadedDialog()
+void UpdateDownloadedDialog::accept()
 {
-    delete _ui;
+    Q_EMIT accepted();
+    close();
```
Comment (TheOneRing) [inline src/gui/updater/updatedownloadeddialog.cpp:50]: We should rather delete the class in the caller than rely on delete on close.

**Review by TheOneRing — CHANGES_REQUESTED** [review]
Looks great, but please take a look on my comments.

File: src/gui/updater/nourlwidget.h (hunk 2 above)
Comment (erikjv) [reply src/gui/updater/nourlwidget.h:31]: It comes from the method that previously showed the dialog: `WindowsUpdater::showNoUrlDialog`. It's called from `WindowsUpdater::versionInfoArrived`, when the info has an empty URL. 🤷‍♂️

Comment (erikjv) [reply src/gui/updater/nourlwidget.h:31]: Rename to `NewVersionAvailableDialog`, same for the calling method.

File: src/gui/updater/updatedownloadeddialog.cpp (hunk 3 above)
Comment (erikjv) [reply src/gui/updater/updatedownloadeddialog.cpp:50]: Do memory management on the creating side, but check the signals.

**Review by TheOneRing — APPROVED** [review]

**Pull request merged**